java/src/main/java/com/google/appengine/tools/pipeline/impl/servlets/TaskHandler.java

```diff
@@ -47,7 +46,6 @@
 public class TaskHandler {
 
   final JobRunServiceComponent component;
-  final RequestUtils requestUtils;
 
   public static final String PATH_COMPONENT = "handleTask";
   public static final String TASK_NAME_REQUEST_HEADER = "X-AppEngine-TaskName";
```

Comment:
:red_circle: watch out, on cloud tasks the retry headers change. Not sure if this reconstruct task is going to work.
https://cloud.google.com/tasks/docs/creating-http-target-tasks#handler

Maybe do a `ObjectUtils.firstNonNull(request.getHeader("X-AppEngine-TaskName"), request.getHeader("X-CloudTasks-TaskName"))`